torch/_inductor/autotune_process.py

```diff
@@ -166,71 +168,179 @@ class BenchmarkRequest:
+                f"InChidProcess {str(self)}: load {load_elapse}, "
+                + f"create tensor {create_tensor_elapse}, bench {bench_elapse}, "
+                + f"collected time {out}"
             )
+        self.cleanup_run_fn()
         return out
 
 
+class TritonBenchmarkRequest(BenchmarkRequest):
+    def __init__(
+        self,
+        kernel_name: str,
+        input_tensor_meta: List[TensorMeta],
+        output_tensor_meta: TensorMeta,
+        extra_args: Dict[str, Any],
+        module_path: str,  # the path of the module defining the triton kernel
+        module_cache_key: str,
+        grid: List[int],
+        num_stages: int,
+        num_warps: int,
+    ):
+        super().__init__(kernel_name, input_tensor_meta, output_tensor_meta, extra_args)
+        self.module_path = module_path
+        self.module_cache_key = module_cache_key
+        self.grid = grid
+        self.num_stages = num_stages
+        self.num_warps = num_warps
+
+    def make_run_fn(
+        self, *input_tensors: torch.Tensor, output_tensor: torch.Tensor
+    ) -> Callable[[], None]:
+        mod = PyCodeCache.load_by_key_path(self.module_cache_key, self.module_path)
+        if DEBUG:
+            print(
+                f"benchmark module key: {self.module_cache_key}, path: {self.module_path}"
+            )
+
+        run_method = getattr(mod, self.kernel_name).run
+
+        return functools.partial(
+            run_method,
+            *input_tensors,
+            output_tensor,
+            *self.extra_args,
+            grid=self.grid,
+            num_stages=self.num_stages,
+            num_warps=self.num_warps,
+            stream=torch.cuda.current_stream().cuda_stream,
+        )
+
+    def __str__(self) -> str:
+        return f"{self.kernel_name=}, {self.module_path=}, {self.module_cache_key=}"
+
+
+class CUDABenchmarkRequest(BenchmarkRequest):
+    def __init__(
+        self,
+        kernel_name: str,
+        input_tensor_meta: List[TensorMeta],
+        output_tensor_meta: TensorMeta,
+        extra_args: Dict[str, Any],
+        source_code: str,
+    ):
+        super().__init__(kernel_name, input_tensor_meta, output_tensor_meta, extra_args)
+        self.source_code = source_code
+        self.workspace_size: int = 0
+        self.workspace: Optional[Tensor] = None
+        self.DLL: Optional[DLLWrapper] = None
+        self.hash_key: str = ""
+        self.source_file: str = ""
+        self.hash_key, self.source_file = CUDACodeCache.write_source_code(
+            self.source_code, "so"
+        )
+
+    def make_run_fn(
+        self, *input_tensors: torch.Tensor, output_tensor: torch.Tensor
+    ) -> Callable[[], None]:
+        self.DLL, self.hash_key, self.source_file = CUDACodeCache.load(
+            self.source_code, "so"
+        )
+        args = [
+            c_void_p(tensor.data_ptr())
+            for tensor in list(input_tensors) + [output_tensor]
+        ]
+        log.debug(
```

Comment:
Init logger at the top of the module first?

Reply:
Fixed.